dd-trace/src/telemetry.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    sync::{Arc, Mutex, OnceLock},
+    time::Duration,
+};
+
+use ddtelemetry::{
+    data::{self},
+    worker::{self, TelemetryWorkerHandle},
+};
+
+use crate::{dd_debug, dd_error, dd_info, Config};
+
+static INIT_TELEMETRY_LOCK: Mutex<()> = Mutex::new(());
+
+static TELEMETRY: OnceLock<Arc<Mutex<Telemetry>>> = OnceLock::new();
+
+pub trait TelemetryHandle: Sync + Send + 'static {
```

Comment:
this trait doesn't need to be public

Reply:
As `init_telemetry` fn is public, `TelemetryHandle` must also be public, isn't it?